lib/ansible/modules/clustering/kubevirt/kubevirt_template.py

```diff
@@ -0,0 +1,349 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2018, KubeVirt Team <@kubevirt>
+# Apache License, Version 2.0
+# (see LICENSE or http://www.apache.org/licenses/LICENSE-2.0)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: kubevirt_template
+
+short_description: Manage KubeVirt templates
+
+description:
+    - Use Openshift Python SDK to manage the state of KubeVirt templates.
+
+version_added: "2.8"
+
+author: KubeVirt Team (@kubevirt)
+
+options:
+    state:
+        description:
+            - Set the to be either I(present) or I(absent).
+        default: "present"
+        choices:
+            - present
+            - absent
+        type: str
+    name:
+        description:
+            - Name of the template.
+        required: true
+        type: str
+    namespace:
+        description:
+            - Namespace where the template exists.
+        required: true
+        type: str
+    objects:
+        description:
+            - List of any valid API objects, such as a I(DeploymentConfig), I(Service), etc. The object
```

Comment:
What kind of template management is this supposed to be? I see you specifically mention DepoymentConfig, Service... all of which are pure kubernetes concepts. A little bellow you added default_disk, nic, volume parameters that are only relevant for VM templates and actually do not make any sense (as we ship them within the Template object).

Reply:
Well, it should support templates of both Openshift/k8s objects and kubevirt, so the description is general, I will improve.